src/components/forms/CorpusForm.tsx

```diff
@@ -0,0 +1,499 @@
+          </FormControl>
+          <Controller
+            control={control}
+            name='corpus_type_name'
+            render={({ field }) => (
+              <FormControl
+                as='fieldset'
+                isRequired
+                isInvalid={!!errors.corpus_type_name}
+              >
+                <FormLabel as='legend'>Corpus Type Name</FormLabel>
+                <div data-testid='corpus-type-select'>
+                  <CRSelect
+                    chakraStyles={chakraStylesSelect}
+                    isClearable={true}
+                    isMulti={false}
+                    isSearchable={true}
+                    options={
+                      uniqueCorpusTypes.map((ct) => ({
+                        label: ct.name,
+                        value: ct.name,
+                      })) || []
+                    }
+                    isDisabled={!!loadedCorpus}
+                    {...field}
+                  />
+                </div>
+                <FormErrorMessage>
+                  {errors.corpus_type_name?.message}
+                </FormErrorMessage>
+              </FormControl>
+            )}
+          />
+          {loadedCorpus && (
+            <FormControl
+              isRequired
+              isInvalid={!!errors.corpus_type_description}
+            >
+              <FormLabel>
+                Corpus Type Description
+                <Tooltip label='Updating this will also apply this change to all other corpora of this type'>
+                  <Icon as={InfoOutlineIcon} ml={2} cursor='pointer' />
+                </Tooltip>
+              </FormLabel>
+              <Textarea
+                height={'100px'}
+                bg='white'
+                {...register('corpus_type_description')}
+              />
+            </FormControl>
+          )}
+
+          <Controller
+            control={control}
+            name='organisation_id'
+            render={({ field }) => (
+              <FormControl
+                as='fieldset'
+                isRequired
+                isInvalid={!!errors.organisation_id}
+              >
+                <FormLabel as='legend'>Organisation</FormLabel>
+                <div data-testid='organisation-select'>
+                  <CRSelect
+                    chakraStyles={chakraStylesSelect}
+                    isClearable={true}
+                    isMulti={false}
+                    isSearchable={true}
+                    options={
+                      Array.from(
+                        new Set(
+                          config?.corpora?.map((corpus) => ({
+                            label: corpus.organisation?.name,
+                            value: corpus.organisation?.id,
+                          })),
+                        ),
+                      ).map((org) => ({
+                        label: org.label,
+                        value: org.value,
+                      })) || []
```

Comment:
Is this `|| []` redundant as this would always be at least an `[]`

Reply:
Good spot!